dev/conditions.h

```diff
@@ -1171,12 +1172,50 @@ namespace sqlite_orm {
         return {std::move(l), std::move(r)};
     }
 
+#if __cplusplus >= 201703L  // use of C++17 or higher
+    /**
+     *  integral_constant<unsigned int, N> from numeric literal.
+     *  E.g. 1_nth_col, 2_nth_col
+     *  
+     *  @note Design desicion:
+     *  A user-defined numeric literal is the C++ way to capture a numeric literal as a compile-time value,
+     *  with the intention to form a literal positional ordinal in the SQL statement.
+     *  It isn't possible with built-in literals.
+     *  The user-defined literal might seem a bit inconvenient at first,
+     *  however has the advantage of giving the literal contextual meaning.
+     */
+    template<char... Chars>
+    [[nodiscard]] SQLITE_ORM_CONSTEVAL auto operator"" _nth_col() {
+        constexpr auto n =
+            internal::nth_constant<internal::n_from_literal(std::make_index_sequence<sizeof...(Chars)>{}, Chars...)>{};
+        static_assert(n > 0u, "Column number must be greater than 0.");
+        return n;
+    }
+#endif
+
     /**
      * ORDER BY column
-     * Example: storage.select(&User::name, order_by(&User::id))
+     * Examples:
+     * storage.select(&User::name, order_by(&User::id))
+     * storage.select(&User::name, order_by(1_nth_col))
      */
     template<class O>
     internal::order_by_t<O> order_by(O o) {
+        static_assert(!std::is_arithmetic<O>::value && !std::is_base_of<std::string, O>::value &&
```

Comment:
actually one can bind any custom type using custom type binding technique. E.g. any custom type like `QInt`, `QString` or whatever but he/she will not be able to use these types because you prohibited it here. Assuming literals is not what ORM should do because it looks like an artificial prohibiting of some functional

Reply:
No, the static assert does not prohibit custom types. Only arithmetic and strings.

Reply:
It should actually ensure it's not a bindable `static_assert(!is_bindable_v<O>, "")`.